src/workflows-app/SubmissionConfig.test.js

```diff
@@ -285,3 +302,1391 @@ describe('BaseSubmissionConfig gets proxy urls from Leo', () => {
+    expect(barRows.length).toBe(3);
+    const barHeaders = within(barRows[0]).queryAllByRole('columnheader');
+    expect(barHeaders.length).toBe(4);
+    within(barHeaders[1]).getByText('ID');
+    // even though both 'FOO' and 'BAR' data tables have 'ID' columns their widths can be different
+    expect(getComputedStyle(barHeaders[1]).width).toBe('300px'); // initial column width
+
+    // ** ACT **
+    // simulate user resizing the column 'ID' for data table 'BAR'
+    const barDraggableIcon = barHeaders[1].querySelector("[data-icon='columnGrabber']");
+    fireEvent.mouseDown(barDraggableIcon);
+    fireEvent.mouseMove(barDraggableIcon, { clientX: 50, clientY: 0 }); // user moves the icon 50px to right
+    fireEvent.mouseUp(barDraggableIcon);
+
+    // ** ASSERT **
+    // new width of column 'ID' for data table 'BAR' should be 350
+    expect(getComputedStyle(barHeaders[1]).width).toBe('350px');
+
+    // ** ACT **
+    // Change Data Table back to 'FOO'
+    const dropdown2 = await screen.findByLabelText('Select a data table');
+    await act(async () => {
+      await selectEvent.select(dropdown2, ['FOO']);
+    });
+
+    // ** ASSERT **
+    // verify that the width of column 'ID' has been preserved from previous resizing
+    const fooRows2 = within(table).queryAllByRole('row');
+    const fooHeaders2 = within(fooRows2[0]).queryAllByRole('columnheader');
+    expect(getComputedStyle(fooHeaders2[1]).width).toBe('500px');
+  });
+
+  it('when records are selected, should display modal when Submit button is clicked', async () => {
+    const mockRunSetResponse = jest.fn(() => Promise.resolve(runSetResponse));
+    const mockMethodsResponse = jest.fn(() => Promise.resolve(methodsResponse));
+    const mockSearchResponse = jest.fn((_root, _instanceId, recordType) => Promise.resolve(searchResponses[recordType]));
+    const mockTypesResponse = jest.fn(() => Promise.resolve(typesResponse));
+    const mockWdlResponse = jest.fn(() => Promise.resolve('mock wdl response'));
+
+    Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runSets: {
+            getForMethod: mockRunSetResponse,
+          },
+          methods: {
+            getById: mockMethodsResponse,
+          },
+        },
+        WorkspaceData: {
+          queryRecords: mockSearchResponse,
+          describeAllRecordTypes: mockTypesResponse,
+        },
+        WorkflowScript: {
+          get: mockWdlResponse,
+        },
+      };
+    });
+
+    // ** ACT **
+    render(
+      h(BaseSubmissionConfig, {
+        methodId: '123',
+        name: 'test-azure-ws-name',
+        namespace: 'test-azure-ws-namespace',
+        workspace: mockAzureWorkspace,
+      })
+    );
+
+    await waitFor(() => {
+      expect(mockRunSetResponse).toHaveBeenCalledTimes(1);
+    });
+
+    const checkboxes = screen.getAllByRole('checkbox');
+    const checkbox = checkboxes[1];
+    fireEvent.click(checkbox);
```

Comment:
Let's prefer `userEvent` over `fireEvent` wherever possible as it's more realistic.
https://testing-library.com/docs/user-event/intro#differences-from-fireevent